sdk/storage/src/blobs/blob_client.cpp

```diff
@@ -74,20 +76,37 @@ namespace Azure { namespace Storage { namespace Blobs {
       const std::string& blobUri,
       std::shared_ptr<SharedKeyCredential> credential,
       const BlobClientOptions& options)
-      : BlobClient(blobUri, options)
+      : m_blobUrl(blobUri)
   {
-    // not implemented yet
-    unused(credential, options);
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonRequestPolicy>());
+    policies.emplace_back(std::make_unique<SharedKeyPolicy>(credential));
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
```

Comment:
Consider following scenarios:
1. Customer wants to log all the request headers/bodies for debugging purposes before the request is send and after the response is received.
2. Customer wants to replace the HTTP transport policy.
How to support those scenarios?

Reply:
1. We cannot do that for now.

2. Customer adds another HTTP transport policy, so this built-in one will be discarded.

The current pipeline implementation is not powerful enough, but should be enough for the first release.

Reply:
I added an item on the OneNote page